### PR TITLE
Remove the version constraint of `fakeredis`

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

Revert [this hotfix](https://github.com/optuna/optuna/pull/3549) for fakeredis because [`fakeredis` 1.7.5](https://github.com/dsoftwareinc/fakeredis-py/releases/tag/v1.7.5), which fixes the issue, has been released.

## Description of the changes

Depending on the python version, we need to set the version constraint of fakeredis:

- `fakeredis<=1.7.1` if python 3.6 or 3.7 
- `fakeredis` otherwise

### CODE REVIEW

**Comment by nzw0301** [comment]
Hmm, fakeredis 1.7.5 drops python 3.7 supports.

**Comment by c-bata** [comment]
@toshihikoyanase Could you review this PR?

**Review by c-bata — APPROVED** [review]
LGTM 👍

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR. I have a question about the fakeredis version for Python 3.7.

File: setup.py [hunk 1]
```diff
@@ -94,8 +94,8 @@ def get_extras_require() -> Dict[str, List[str]]:
         "testing": [
             "chainer>=5.0.0",
             "cma",
-            # TODO(HideakiImamura): Remove the version constraint after the next release.
-            "fakeredis<=1.7.1",
+            "fakeredis<=1.7.1; python_version<'3.8'",
+            "fakeredis ; python_version>='3.8'",
```
[the remark this answers is not in the document]
Comment (nzw0301) [reply setup.py:98]: Thank you for pointing out it. I didn't see the pypi, but according to the difference between [v1.7.4 and v1.7.5](https://github.com/dsoftwareinc/fakeredis-py/compare/v1.7.4...v1.7.5), fakeredis does not test python 3.7 anymore on its CI jobs.

[the remark this answers is not in the document]
Comment (nzw0301) [reply setup.py:98]: Let's send a PR to know the situation -> filed https://github.com/dsoftwareinc/fakeredis-py/pull/8.

[the remark this answers is not in the document]
Comment (nzw0301) [reply setup.py:98]: The PR has been merged. I suppose the change makes sense.

[the remark this answers is not in the document]
Comment (toshihikoyanase) [reply setup.py:98]: Thanks!

**Review by toshihikoyanase — APPROVED** [review]
LGTM!

**Pull request merged**